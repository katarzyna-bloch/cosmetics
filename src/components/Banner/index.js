import styled from 'styled-components'
import React from 'react'
import { Link as RouterLink } from 'react-router-dom'
import PropTypes from 'prop-types'

import media from '../../utils/media'
import { getImageBannersPath  } from '../../utils/images'

const Link = styled(RouterLink)`
  text-decoration: none;
  color: black;
`

const Wrapper = styled.div`
  position: relative;
`

const Image = styled.img`
  width: 100%;
  height: 500px;
  object-fit: cover;
`

const Text = styled.div`
  font-family: cursive;
  font-weight: 600;
  font-size: 25px;
  position: absolute;
  background-color: whitesmoke;
  padding: 2px;
  bottom: 30px;
  left: 0;
  right: 0;
  text-align: center;
  margin: 20px;
 
  ${media.tablet} {
    margin: 10px;
    font-size: 20px;
  }
`

export const Banner = ({ banner: { img, url, text } }) => (
  <Wrapper>
    <Link to={url}>
      <Image src={getImageBannersPath(img)} alt="banners" />
      <Text>{text}</Text>
    </Link> 
  </Wrapper>
)

Banner.propTypes = {
  banner: PropTypes.shape({
    img: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  }).isRequired,
}

export default Banner
