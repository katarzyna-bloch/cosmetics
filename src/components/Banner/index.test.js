import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Banner from '.'

const props = {
  id: '3',
  img: 'makeup.jpg',
  url: '/categories/makeup', 
  text: 'Różnorodne rabaty dla Ciebie. Zobacz wybrane produkty w super cenach!',
}

it('should render correctly', () => {
  const wrapped = shallow(<Banner {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
