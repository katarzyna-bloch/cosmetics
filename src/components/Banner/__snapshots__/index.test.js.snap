// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div>
  <Styled(Link)
    to="/categories/makeup"
  >
    <styled.img
      alt="banners"
      src="/images/banners/makeup.jpg"
    />
    <styled.div>
      Różnorodne rabaty dla Ciebie. Zobacz wybrane produkty w super cenach!
    </styled.div>
  </Styled(Link)>
</styled.div>
`;
